### ISSUE TITLE
FHIR lookup for seeding the cache with VA Profile IDs -  e2e testing

### ISSUE THREAD

**Issue by cris-oddball** [department-of-veterans-affairs/notification-api#1683]
## User Story - Business Need

_**Required.** Please note you can have more than one story if applicable. If this is technical work achieving a larger initiative, the user story should represent that piece of the initiative, so it’s clear what problem we are solving._

We want to perform e2e testing of the BIP workflow so that we can be assured that the pieces all work together. In this ticket, we obtain the VAProfile IDs from Participant IDs collected from the PERF consumer in Jan/Feb 2024

See  IDs identified from Perf in the team repo [here](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/Quality%20Assurance/test_users_kafka_e2e_test.md).

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)
**As a**   VANotify engineer
**I want**    to configure a testing setup for the kafka integration
**So that**  i can be assured that it works prior to sending real data through the system.

### Additional Info and Resources

- Rather than involving other teams when we have control over the vaprofile cache (a table in the notification-api database), we want to seed the cache with certain IDs set to have the specified communication item preferences for Comp & Pen delivery where
  - "communicationItemId" is 5
  -  "communicationChannelId" is 1
  -  "allowed" is  true
- Participant IDs were collected from the Perf consumer for the January and February 2024 events. These IDs need to be looked up in FHIR to obtain the VAProfile IDs so that the cache can be populated
  - FHIR info: https://ps.dev.iam.va.gov/iam/mpi/fhir/r4/ 
  - We also do this lookup in our codebase
  - A handy script should be created to just loop over the IDs in the list
- Not every ID should be included in the cache so that we can ensure that if a Participant ID is received but we do not find the corresponding VAProfile ID in the cache, then we don't send the notification.
- Work with QA to determine which IDs to exclude from the cache.

## Engineering Checklist

- [x] See the [linked document](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/Quality%20Assurance/test_users_kafka_e2e_test.md) with the known good UAT Participant IDs; run these through FHIR to get the corresponding VAProfile IDs (ideally using a script, which is shared with QA)
- [x] Ensure the VA Profile has `^A` at the end, for `Active` according to MPI
- [x] Document all IDs in the team repo in the linked document. For the provided Participant IDs, update in the table which of those IDs has a corresponding VA Profile ID.

## Acceptance Criteria

- Documentation is provided to QA as outlined in the Engineering Checklist

## QA Considerations
- Review the documentation

## Out of Scope
- Using the producer
- Adding IDs to the cache (see #1619)

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1683?planning-poker) with Zenhub @cris-oddball @EvanParish @k-macmillan @kalbfled @ldraney @MackHalliday @mchlwellman

**Comment by kalbfled** [comment]
This can be implemented by cobbling together code we've already written with little modification.  The MPI lookup lambda can be modified to lookup participant_id-->profile_id, and then the lambda can make a request to our existing VA Profile cache lambda.

**Comment by kalbfled** [comment]
I e-mailed a list of mapping to @cris-oddball.

To execute the code that makes the MPI requests, I used AWS Lambda console to modify the handler for project-perf-bip-msg-mpi-lookup-lambda to call the code in the attached Python file.  This was convenient because it allowed me to leverage the existing code that creates the SSL context needed to authenticate with MPI.

Initially, I intended to write a stand-alone solution that I could run from GFE, but my VA issued laptop does not have Python installed.  Then I attempted to create a new lambda function to run the code, but I don't have permission in AWS.

After e-mailing the results, I deleted the changes to the Perf lambda.

[participant_to_profile.zip](https://github.com/department-of-veterans-affairs/notification-api/files/14727690/participant_to_profile.zip)
